engine/hid/src/glfw/hid_native.cpp

```diff
@@ -125,43 +123,54 @@ namespace dmHID
         // Update keyboard
         if (!context->m_IgnoreKeyboard)
         {
-            // TODO: Actually detect if the keyboard is present
-            context->m_KeyboardConnected = 1;
-            for (uint32_t i = 0; i < MAX_KEY_COUNT; ++i)
+            for (uint32_t k = 0; k < MAX_KEYBOARD_COUNT; ++k)
             {
-                uint32_t mask = 1;
-                mask <<= i % 32;
-                int key = glfwGetKey(i);
-                if (key == GLFW_PRESS)
-                    context->m_KeyboardPacket.m_Keys[i / 32] |= mask;
-                else
-                    context->m_KeyboardPacket.m_Keys[i / 32] &= ~mask;
+                Keyboard* keyboard = &context->m_Keyboards[k];
+                keyboard->m_Connected = 1; // TODO: Actually detect if the keyboard is present
+
+                bool pressed = false;
+                for (uint32_t i = 0; i < MAX_KEY_COUNT; ++i)
+                {
+                    uint32_t mask = 1;
+                    mask <<= i % 32;
+                    int state = glfwGetKey(i);
+                    if (state == GLFW_PRESS)
+                        keyboard->m_Packet.m_Keys[i / 32] |= mask;
+                    else
+                        keyboard->m_Packet.m_Keys[i / 32] &= ~mask;
+                }
             }
         }
 
         // Update mouse
         if (!context->m_IgnoreMouse)
         {
-            // TODO: Actually detect if the keyboard is present, this is important for mouse input and touch input to not interfere
-            context->m_MouseConnected = 1;
-            MousePacket& packet = context->m_MousePacket;
-            for (uint32_t i = 0; i < MAX_MOUSE_BUTTON_COUNT; ++i)
+            for (uint32_t m = 0; m < MAX_MOUSE_COUNT; ++m)
             {
-                uint32_t mask = 1;
-                mask <<= i % 32;
-                int button = glfwGetMouseButton(i);
-                if (button == GLFW_PRESS)
-                    packet.m_Buttons[i / 32] |= mask;
-                else
-                    packet.m_Buttons[i / 32] &= ~mask;
-            }
-            int32_t wheel = glfwGetMouseWheel();
-            if (context->m_FlipScrollDirection)
-            {
-                wheel *= -1;
+                Mouse* mouse = &context->m_Mice[m];
+                // TODO: Actually detect if the keyboard is present,
```

Comment:
You mean mouse?

Reply:
Ah, saw that this was an existing comment. I don't understand what the keyboard has to do with mouse and touch though...